### PR TITLE
Geomerged

### PR DESCRIPTION
**Pull request by pkofod**
Merged conflict resolved version of https://github.com/JuliaNLSolvers/LsqFit.jl/pull/106

@Magalame

### CODE REVIEW

File: src/curve_fit.jl [hunk 1]
```diff
@@ -29,18 +29,25 @@ function lmfit(f, g, p0::AbstractArray, wt::AbstractArray; kwargs...)
...
-function lmfit(f!, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; autodiff = :finite, kwargs...)
-    autodiff = autodiff == :forwarddiff ? :forward : autodiff
-    R = OnceDifferentiable(f!, p0, similar(r); inplace = true, autodiff = autodiff)
+function lmfit(f, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; inplace=false, autodiff = :finite, kwargs...)
+    R = OnceDifferentiable(f, p0, similar(r); inplace = true, autodiff = autodiff)
     lmfit(R, p0, wt; kwargs...)
 end
 
+#experimental geo, I let the `inplace` machinery but it isn't avilable yet
+function lmfit(f, g, avv!, p0::AbstractArray, wt::AbstractArray; inplace = false, kwargs...)
```
Comment (Magalame) [inline src/curve_fit.jl]: This function is actually not used anymore, I forgot to remove it, sorry! `avv!` is now passed as a `kwarg`.

File: src/curve_fit.jl [hunk 2]
```diff
@@ -29,18 +29,25 @@ function lmfit(f, g, p0::AbstractArray, wt::AbstractArray; kwargs...)
 end
 
 #for inplace f and inplace g
-function lmfit(f!, g!, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; kwargs...)
-    R = OnceDifferentiable(f!, g!, p0, similar(r); inplace = true)
+function lmfit(f, g, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; inplace=false, kwargs...)
```
Comment (Magalame) [inline src/curve_fit.jl]: It'd probably be nice to actually keep `f!` and `g!`, also we don't need the `inplace=false` anymore

File: src/curve_fit.jl [hunk 3]
```diff
@@ -29,18 +29,25 @@ function lmfit(f, g, p0::AbstractArray, wt::AbstractArray; kwargs...)
...
+    R = OnceDifferentiable(f, g, p0, similar(r); inplace = true)
     lmfit(R, p0, wt; kwargs...)
 end
 
 #for inplace f only
-function lmfit(f!, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; autodiff = :finite, kwargs...)
-    autodiff = autodiff == :forwarddiff ? :forward : autodiff
```
Comment (Magalame) [inline src/curve_fit.jl]: We should probably keep this line

File: src/curve_fit.jl [hunk 4]
```diff
@@ -29,18 +29,25 @@ function lmfit(f, g, p0::AbstractArray, wt::AbstractArray; kwargs...)
...
-    R = OnceDifferentiable(f!, p0, similar(r); inplace = true, autodiff = autodiff)
+function lmfit(f, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; inplace=false, autodiff = :finite, kwargs...)
```
Comment (Magalame) [inline src/curve_fit.jl]: Same comment as above: we should keep `f!`, and remove `inplace=false`

**Comment by Magalame** [comment]
bump

File: src/curve_fit.jl (hunk 3 above)
Comment (pkofod) [reply src/curve_fit.jl]: why? this should be handled by NLSolversBase

**Comment by pkofod** [comment]
I think it's ready now, no?

**Comment by Magalame** [comment]
Seems like it! I'm really happy we got it done eventually

**Comment by pkofod** [comment]
> Seems like it! I'm really happy we got it done eventually

yeah, that's on me. Seems like I removed a method too much, or the inplace keyword is called somewhere where it shouldn't. Will fix :)

**Comment by pkofod** [comment]
Thanks for the work :)

**Comment by Magalame** [comment]
🎉

**Pull request merged**